### ISSUE TITLE
Exclude paramiko module from ansible build?

### ISSUE THREAD

**Issue by ghost** [ansible/ansible#11729]
Hi,

I was wondering if there could be any potential problems of excluding paramiko from Ansible? According to http://www.ansible.com/blog/2014/01/15/ssh-connection-upgrades-coming-in-ansible-1-5 you used it in the past for ControlPersist which is a feature available in OpenSSH 5.6 (released maybe a year ago) and as such appears redundant. You also list some rationales for having it in "How do I get ansible to reuse connections, enable Kerberized SSH, or have Ansible pay attention to my local SSH config file?"
https://docs.ansible.com/ansible/faq.html which seems to me to be a legacy issue only. Furthermore these explanations:

"Paramiko is great for starting out, but the OpenSSH type offers many advanced options." 

This line doesn't make any sense. Great for starting what out?

"You will want to run Ansible from a machine new enough to support ControlPersist, if you are using this connection type." 

"A machine" or openssh version? 

"You can still manage older clients. If you are using RHEL 6, CentOS 6, SLES 10 or SLES 11 the version of OpenSSH is still a bit old, so consider managing from a Fedora or openSUSE client even though you are managing older nodes, or just use paramiko."

How about you avoid all this confusing nonsense and just recommend using the latest openssh?

Caveat emptor: I don't know anything about https://github.com/paramiko/paramiko but it has not been updated since April (stable?) and it has a large number of bug and pull requests that seems like a lot of work for one guy to be working part time let alone full time. 

Basically I want to make sure I can exclude this from my ansible builds because adding unnecessary code to my environment adds unnecessary risk. 

If this is the case then you should also remove it from the list of required modules on https://docs.ansible.com/ansible/intro_installation.html and list it as optional.

Thanks!

**Comment by bcoca** [comment]
paramiko is still needed for those that have older versions and cannot upgrade as it is much faster than ssh w/o ControlMaster. It is also needed for systems that do not work well with sshpass (i.e. OS X, BSDs) so we cannot remove it. 

If you are sure you don't need it you can take it out of your builds as well as the paramiko_ssh connection plugin which is what uses it directly.

That said I'm closing this ticket as we will not be removing it due to the need of many of ansible users to have this option.
